tests/functional/aws-node-sdk/test/object/encryptionHeaders.js

```diff
@@ -0,0 +1,343 @@
+                        Bucket: bucket,
+                        Key: object,
+                    };
+                    if (target.algo) {
+                        params.ServerSideEncryption = target.algo;
+                    }
+                    if (target.masterKeyId) {
+                        params.SSEKMSKeyId = kmsKeyId;
+                    }
+                    s3.createMultipartUpload(params, (error, resp) => {
+                        assert.ifError(error);
+                        const { UploadId } = resp;
+                        const partParams = {
+                            UploadId,
+                            Body: 'somedata',
+                            Bucket: bucket,
+                            Key: object,
+                            PartNumber: 1,
+                        };
+                        s3.uploadPart(partParams, error => {
+                            assert.ifError(error);
+                            done();
+                        });
+                    });
+                });
+
+                it('should copy and encrypt a mpu part', done => {
+                    const sourceParams = {
+                        Bucket: bucket,
+                        Key: object,
+                    };
+                    s3.createMultipartUpload(sourceParams, (error, resp) => {
+                        assert.ifError(error);
+                        const { UploadId: sourceUploadId } = resp;
+                        const sourcePartParams = {
+                            UploadId: sourceUploadId,
+                            Body: 'somedata',
+                            Bucket: bucket,
+                            Key: object,
+                            PartNumber: 1,
+                        };
+                        s3.uploadPart(sourcePartParams, error => {
+                            assert.ifError(error);
+                            const targetParams = {
+                                Bucket: bucket,
+                                Key: object2,
+                            };
+                            if (target.algo) {
+                                targetParams.ServerSideEncryption = target.algo;
+                            }
+                            if (target.masterKeyId) {
+                                targetParams.SSEKMSKeyId = kmsKeyId;
+                            }
+                            s3.createMultipartUpload(targetParams, (error, resp) => {
+                                const { UploadId: targetUploadId } = resp;
+                                const targetPartParams = {
+                                    UploadId: targetUploadId,
+                                    Body: 'somedata',
+                                    Bucket: bucket,
+                                    Key: object2,
+                                    PartNumber: 1,
+                                };
+                                s3.uploadPart(targetPartParams, error => {
```

Comment:
It seems to me we are not copying anything here but creating a new target part with new data, maybe `uploadPartCopy` was intended?